trio/_sync.py

```diff
@@ -89,20 +96,20 @@ def statistics(self):
 
 class AsyncContextManagerMixin:
     @enable_ki_protection
-    async def __aenter__(self):
-        await self.acquire()
+    async def __aenter__(self) -> None:
+        await self.acquire()  # type: ignore[attr-defined]
 
     @enable_ki_protection
-    async def __aexit__(self, *args):
-        self.release()
+    async def __aexit__(self, *args: object) -> None:
+        self.release()  # type: ignore[attr-defined]
```

Comment:
slightly illegal `type: ignore`s, but would probably require some significant refactoring to get rid of.